### PR TITLE
Issue 1

### PR DESCRIPTION
**Pull request by Wagnesio**

### CODE REVIEW

File: TKCurrencyTextField/Classes/TKCurrencyTextField.swift [hunk 1]
```diff
@@ -54,21 +54,40 @@ import UIKit
...
+        super.awakeFromNib()
+        initTextField()
+    }
+    
     private func initTextField(){
         let currencyHandler: ((_ textField: UITextField, _ range: NSRange, _ replacementString: String) -> ()) = {
             (textField, range, replacementString) in
             let nsString: NSString = textField.text as NSString? ?? ""
             let text = nsString.replacingCharacters(in: range, with: replacementString)
-            let cleanNumericString: String = text.onlyNumberString
```
Comment (willwfsp) [inline TKCurrencyTextField/Classes/TKCurrencyTextField.swift:62]: Prefer compact code and let the compiler infer the type for constants or variables of single instances.

Ref: [Type Inference](https://github.com/raywenderlich/swift-style-guide#type-inference)

File: TKCurrencyTextField/Classes/TKCurrencyTextField.swift [hunk 2]
```diff
@@ -54,21 +54,40 @@ import UIKit
...
+            
+            //Remove mask from newText
+            var cleanNumericString: String = text.onlyNumberString
```
Comment (willwfsp) [inline TKCurrencyTextField/Classes/TKCurrencyTextField.swift:69]: Prefer compact code and let the compiler infer the type for constants or variables of single instances.

Ref: [Type Inference](https://github.com/raywenderlich/swift-style-guide#type-inference)

File: TKCurrencyTextField/Classes/TKCurrencyTextField.swift [hunk 3]
```diff
@@ -54,21 +54,40 @@ import UIKit
...
             var textFieldNumber: Double = 0.0
             
+            //Check if currency symbol was replaced
+            // In case currency symbol was replaced, erase last character
+            let formatter:NumberFormatter = NumberFormatter()
+            formatter.locale = self.locale
+            formatter.numberStyle = .currency
+            if !text.contains(formatter.currencySymbol) {
+                let lastIndex: String.Index = cleanNumericString.endIndex;
+               cleanNumericString = cleanNumericString.substring(to: cleanNumericString.index(before: lastIndex))
```
Comment (willwfsp) [inline TKCurrencyTextField/Classes/TKCurrencyTextField.swift]: Please reident your code properly. If possible, use 2 spaces identation. Use [Swift Lint](https://github.com/realm/SwiftLint) to help you on this

File: TKCurrencyTextField/Classes/TKCurrencyTextField.swift [hunk 4]
```diff
@@ -93,6 +112,13 @@ import UIKit
         setCursorOriginalPosition(cursorOffset, oldTextFieldLength: textFieldLength)
     }
     
+    
+    override open func closestPosition(to point: CGPoint) -> UITextPosition? {
+        let beginning:UITextPosition = self.beginningOfDocument
```
Comment (willwfsp) [inline TKCurrencyTextField/Classes/TKCurrencyTextField.swift:117]: Same tip here. Prefer compact code and let the compiler infer the type for constants or variables of single instances.

Please verify it for whole project
Ref: [Type Inference](https://github.com/raywenderlich/swift-style-guide#type-inference)

File: TKCurrencyTextField/Classes/TKCurrencyTextField.swift [hunk 5]
```diff
@@ -54,21 +54,40 @@ import UIKit
         initTextField()
     }
     
+    open override func awakeFromNib() {
+        super.awakeFromNib()
+        initTextField()
+    }
+    
     private func initTextField(){
         let currencyHandler: ((_ textField: UITextField, _ range: NSRange, _ replacementString: String) -> ()) = {
```
Comment (willwfsp) [inline TKCurrencyTextField/Classes/TKCurrencyTextField.swift:63]: Underscore charater is not necessary. You can use this form:

```swift
((UITextField, NSRange, String) -> ())
```

If type isn't clear itself, create `typealiases`:

```swift
typealias ReplacementString = String

...

((UITextField, NSRange, ReplacementString) -> ())
```

**Review by damboscolo — APPROVED** [review]

**Comment by willwfsp** [comment]
@damboscolo pull request can be merged only after my approval. See the comments below about the code style.

**Comment by Edudjr** [comment]
Hey guys, remember to close the branch when it is merged to keep the repo clean! 
Nice work btw :)

ps: does this merge fix the issue https://github.com/tokenlab/TKCurrencyTextField/issues/4 ?

**Pull request merged**